cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -438,56 +462,102 @@ std::shared_ptr<AesDecryptor> AesDecryptor::Make(
   return decryptor;
 }
 
-int AesDecryptor::CiphertextSizeDelta() { return impl_->ciphertext_size_delta(); }
-
-int AesDecryptor::AesDecryptorImpl::GcmDecrypt(const uint8_t* ciphertext,
-                                               int ciphertext_len, const uint8_t* key,
-                                               int key_len, const uint8_t* aad,
-                                               int aad_len, uint8_t* plaintext) {
-  int len;
-  int plaintext_len;
+int AesDecryptor::PlaintextLength(int ciphertext_len) const {
```

Comment:
Should we replace `int` with `int32_t` to be more portable? Same for functions below. Of course this can be a followup change.

Reply:
Yes it would make sense to do this as a follow up to avoid changing too much in this PR, I've made https://github.com/apache/arrow/issues/43141 for this